Verification crashes for `v0.1` bundles missing checkpoints

## Description

I observed this behavior while trying to cross-check `sigstore-python`'s handling of `v0.1` bundles with other clients.

Reproduction steps:

```
wget https://www.python.org/ftp/python/3.12.5/Python-3.12.5.tgz
wget https://www.python.org/ftp/python/3.12.5/Python-3.12.5.tgz.sigstore
sigstore-go -artifact Python-3.12.5.tgz -expectedSAN 'thomas@python.org' Python-3.12.5.tgz.sigstore 
```

Running that fails with:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x30 pc=0x10293ec40]

goroutine 1 [running]:
github.com/sigstore/sigstore-go/pkg/tlog.ParseEntry(0x140003c8f50)
	/Users/william/go/pkg/mod/github.com/sigstore/sigstore-go@v0.5.1/pkg/tlog/entry.go:132 +0x500
github.com/sigstore/sigstore-go/pkg/bundle.(*ProtobufBundle).TlogEntries(0x140003d4860)
	/Users/william/go/pkg/mod/github.com/sigstore/sigstore-go@v0.5.1/pkg/bundle/bundle.go:252 +0xb0
github.com/sigstore/sigstore-go/pkg/bundle.(*ProtobufBundle).validate(0x140003d4860)
	/Users/william/go/pkg/mod/github.com/sigstore/sigstore-go@v0.5.1/pkg/bundle/bundle.go:84 +0x100
github.com/sigstore/sigstore-go/pkg/bundle.(*ProtobufBundle).UnmarshalJSON(0x140003d4860, {0x14000389500, 0x14ad, 0x14ae})
	/Users/william/go/pkg/mod/github.com/sigstore/sigstore-go@v0.5.1/pkg/bundle/bundle.go:189 +0xa4
github.com/sigstore/sigstore-go/pkg/bundle.LoadJSONFromPath({0x16d997928, 0x33})
	/Users/william/go/pkg/mod/github.com/sigstore/sigstore-go@v0.5.1/pkg/bundle/bundle.go:170 +0x8c
main.run()
	/Users/william/go/pkg/mod/github.com/sigstore/sigstore-go@v0.5.1/cmd/sigstore-go/main.go:94 +0x58
main.main()
	/Users/william/go/pkg/mod/github.com/sigstore/sigstore-go@v0.5.1/cmd/sigstore-go/main.go:87 +0x1c
```

From a quick triage, that looks like it fails on this `swag.String` ctor:

https://github.com/sigstore/sigstore-go/blob/004c4250f082b69a03df43a95930fd72cef63549/pkg/tlog/entry.go#L132

...which I *suspect* fails because `InclusionProof.Checkpoint` is completely missing from the bundle, which gets silently ignored during unpacking because `protobuf` is very malleable about missing items.

## Version

I tested this with `go install github.com/sigstore/sigstore-go/cmd/sigstore-go@latest`, which I _believe_ should be installing the latest tag (`v0.5.1`).


## Additional context

This is arguably a knock-on bug: the Sigstore bundle in question was generated by `sigstore-python` in the `1.x` series, which didn't include the `checkpoint` field in its bundles (since it wasn't clear from the `v0.1` bundle spec that it was required). 

I'm tracking the associated behavior in sigstore-python here: https://github.com/sigstore/sigstore-python/issues/1088

--- a/pkg/bundle/bundle_test.go
+++ b/pkg/bundle/bundle_test.go
@@ -25,7 +25,6 @@ import (
 
 	protobundle "github.com/sigstore/protobuf-specs/gen/pb-go/bundle/v1"
 	protocommon "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
-	v1 "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
 	rekorv1 "github.com/sigstore/protobuf-specs/gen/pb-go/rekor/v1"
 	_ "github.com/sigstore/rekor/pkg/types/hashedrekord"
 	"github.com/stretchr/testify/require"
@@ -280,7 +279,79 @@ func Test_validate(t *testing.T) {
 							},
 						},
 						Content: &protobundle.VerificationMaterial_PublicKey{
-							PublicKey: &v1.PublicKeyIdentifier{},
+							PublicKey: &protocommon.PublicKeyIdentifier{},
+						},
+					},
+					Content: &protobundle.Bundle_MessageSignature{},
+				},
+			},
+		},
+		{
+			name: "v0.1 with inclusion promise & proof without checkpoint",
+			pb: Bundle{
+				Bundle: &protobundle.Bundle{
+					MediaType: "application/vnd.dev.sigstore.bundle+json;version=0.1",
+					VerificationMaterial: &protobundle.VerificationMaterial{
+						TlogEntries: []*rekorv1.TransparencyLogEntry{
+							{
+								LogIndex: 42,
+								LogId: &protocommon.LogId{
+									KeyId: []byte("deadbeef"),
+								},
+								KindVersion: &rekorv1.KindVersion{
+									Kind:    "hashedrekord",
+									Version: "0.0.1",
+								},
+								IntegratedTime:    1,
+								CanonicalizedBody: canonicalTlogBody,
+								InclusionProof: &rekorv1.InclusionProof{
+									LogIndex: 42,
+									RootHash: []byte("b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c"),
+								},
+								InclusionPromise: &rekorv1.InclusionPromise{
+									SignedEntryTimestamp: []byte("1"),
+								},
+							},
+						},
+						Content: &protobundle.VerificationMaterial_PublicKey{
+							PublicKey: &protocommon.PublicKeyIdentifier{},
+						},
+					},
+					Content: &protobundle.Bundle_MessageSignature{},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "v0.1 with inclusion proof & promise",
+			pb: Bundle{
+				Bundle: &protobundle.Bundle{
+					MediaType: "application/vnd.dev.sigstore.bundle+json;version=0.1",
+					VerificationMaterial: &protobundle.VerificationMaterial{
+						TlogEntries: []*rekorv1.TransparencyLogEntry{
+							{
+								LogIndex: 42,
+								LogId: &protocommon.LogId{
+									KeyId: []byte("deadbeef"),
+								},
+								KindVersion: &rekorv1.KindVersion{
+									Kind:    "hashedrekord",
+									Version: "0.0.1",
+								},
+								IntegratedTime:    1,
+								CanonicalizedBody: canonicalTlogBody,
+								InclusionProof: &rekorv1.InclusionProof{
+									LogIndex:   42,
+									RootHash:   []byte("b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c"),
+									Checkpoint: &rekorv1.Checkpoint{Envelope: "checkpoint"},
+								},
+								InclusionPromise: &rekorv1.InclusionPromise{
+									SignedEntryTimestamp: []byte("1"),
+								},
+							},
+						},
+						Content: &protobundle.VerificationMaterial_PublicKey{
+							PublicKey: &protocommon.PublicKeyIdentifier{},
 						},
 					},
 					Content: &protobundle.Bundle_MessageSignature{},
@@ -308,7 +379,74 @@ func Test_validate(t *testing.T) {
 							},
 						},
 						Content: &protobundle.VerificationMaterial_PublicKey{
-							PublicKey: &v1.PublicKeyIdentifier{},
+							PublicKey: &protocommon.PublicKeyIdentifier{},
+						},
+					},
+					Content: &protobundle.Bundle_MessageSignature{},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "v0.2 with inclusion proof without checkpoint",
+			pb: Bundle{
+				Bundle: &protobundle.Bundle{
+					MediaType: "application/vnd.dev.sigstore.bundle+json;version=0.2",
+					VerificationMaterial: &protobundle.VerificationMaterial{
+						TlogEntries: []*rekorv1.TransparencyLogEntry{
+							{
+								LogIndex: 42,
+								LogId: &protocommon.LogId{
+									KeyId: []byte("deadbeef"),
+								},
+								KindVersion: &rekorv1.KindVersion{
+									Kind:    "hashedrekord",
+									Version: "0.0.1",
+								},
+								IntegratedTime:    1,
+								CanonicalizedBody: canonicalTlogBody,
+								InclusionProof: &rekorv1.InclusionProof{
+									LogIndex: 42,
+									RootHash: []byte("b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c"),
+								},
+							},
+						},
+						Content: &protobundle.VerificationMaterial_PublicKey{
+							PublicKey: &protocommon.PublicKeyIdentifier{},
+						},
+					},
+					Content: &protobundle.Bundle_MessageSignature{},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "v0.2 with inclusion proof with empty checkpoint",
+			pb: Bundle{
+				Bundle: &protobundle.Bundle{
+					MediaType: "application/vnd.dev.sigstore.bundle+json;version=0.2",
+					VerificationMaterial: &protobundle.VerificationMaterial{
+						TlogEntries: []*rekorv1.TransparencyLogEntry{
+							{
+								LogIndex: 42,
+								LogId: &protocommon.LogId{
+									KeyId: []byte("deadbeef"),
+								},
+								KindVersion: &rekorv1.KindVersion{
+									Kind:    "hashedrekord",
+									Version: "0.0.1",
+								},
+								IntegratedTime:    1,
+								CanonicalizedBody: canonicalTlogBody,
+								InclusionProof: &rekorv1.InclusionProof{
+									LogIndex:   42,
+									RootHash:   []byte("b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c"),
+									Checkpoint: &rekorv1.Checkpoint{},
+								},
+							},
+						},
+						Content: &protobundle.VerificationMaterial_PublicKey{
+							PublicKey: &protocommon.PublicKeyIdentifier{},
 						},
 					},
 					Content: &protobundle.Bundle_MessageSignature{},
@@ -337,12 +475,12 @@ func Test_validate(t *testing.T) {
 								InclusionProof: &rekorv1.InclusionProof{
 									LogIndex:   42,
 									RootHash:   []byte("b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c"),
-									Checkpoint: &rekorv1.Checkpoint{},
+									Checkpoint: &rekorv1.Checkpoint{Envelope: "checkpoint"},
 								},
 							},
 						},
 						Content: &protobundle.VerificationMaterial_PublicKey{
-							PublicKey: &v1.PublicKeyIdentifier{},
+							PublicKey: &protocommon.PublicKeyIdentifier{},
 						},
 					},
 					Content: &protobundle.Bundle_MessageSignature{},
@@ -370,7 +508,7 @@ func Test_validate(t *testing.T) {
 								InclusionProof: &rekorv1.InclusionProof{
 									LogIndex:   42,
 									RootHash:   []byte("b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c"),
-									Checkpoint: &rekorv1.Checkpoint{},
+									Checkpoint: &rekorv1.Checkpoint{Envelope: "checkpoint"},
 								},
 							},
 						},
@@ -404,7 +542,7 @@ func Test_validate(t *testing.T) {
 								InclusionProof: &rekorv1.InclusionProof{
 									LogIndex:   42,
 									RootHash:   []byte("b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c"),
-									Checkpoint: &rekorv1.Checkpoint{},
+									Checkpoint: &rekorv1.Checkpoint{Envelope: "checkpoint"},
 								},
 							},
 						},
@@ -818,7 +956,7 @@ func Test_BundleValidation(t *testing.T) {
 					Content:   &protobundle.Bundle_DsseEnvelope{},
 					VerificationMaterial: &protobundle.VerificationMaterial{
 						Content: &protobundle.VerificationMaterial_PublicKey{
-							PublicKey: &v1.PublicKeyIdentifier{},
+							PublicKey: &protocommon.PublicKeyIdentifier{},
 						},
 						TimestampVerificationData: &protobundle.TimestampVerificationData{},
 					},

--- a/pkg/tlog/entry.go
+++ b/pkg/tlog/entry.go
@@ -124,6 +124,13 @@ func ParseEntry(protoEntry *v1.TransparencyLogEntry) (entry *Entry, err error) {
 
 		rootHash := hex.EncodeToString(protoEntry.InclusionProof.RootHash)
 
+		if protoEntry.InclusionProof.Checkpoint == nil {
+			return nil, fmt.Errorf("inclusion proof missing required checkpoint")
+		}
+		if protoEntry.InclusionProof.Checkpoint.Envelope == "" {
+			return nil, fmt.Errorf("inclusion proof checkpoint empty")
+		}
+
 		inclusionProof = &models.InclusionProof{
 			LogIndex:   swag.Int64(protoEntry.InclusionProof.LogIndex),
 			RootHash:   &rootHash,

